packages/fuels-code-gen/src/program_bindings/abigen/configurables.rs

```diff
@@ -0,0 +1,118 @@
+use std::collections::HashSet;
+
+use proc_macro2::{Ident, TokenStream};
+use quote::quote;
+
+use crate::{
+    error::Result,
+    program_bindings::{
+        abi_types::{FullConfigurable, FullTypeDeclaration},
+        resolved_type::{resolve_type, ResolvedType},
+    },
+    utils::safe_ident,
+};
+
+#[derive(Debug)]
+pub(crate) struct ResolvedConfigurable {
+    pub name: Ident,
+    pub ttype: ResolvedType,
+    pub offset: u64,
+}
+
+impl ResolvedConfigurable {
+    pub fn new(
+        configurable: &FullConfigurable,
+        shared_types: &HashSet<FullTypeDeclaration>,
+    ) -> Result<ResolvedConfigurable> {
+        Ok(ResolvedConfigurable {
+            name: safe_ident(&format!("set_{}", configurable.name.to_lowercase())),
+            ttype: resolve_type(&configurable.application, shared_types)?,
+            offset: configurable.offset,
+        })
+    }
+}
+
+pub(crate) fn generate_code_for_configurable_constatnts(
+    name: &Ident,
+    configurables: &[FullConfigurable],
+    shared_types: &HashSet<FullTypeDeclaration>,
+) -> Result<TokenStream> {
+    let resolved_configurables = configurables
+        .iter()
+        .map(|c| ResolvedConfigurable::new(c, shared_types))
+        .collect::<Result<Vec<_>>>()?;
+
+    let config_methods = generate_configurable_impl(name, &resolved_configurables);
+    let from_impl = generate_from_impl(name);
+
+    Ok(quote! {
+        #[derive(Clone, Debug, Default)]
+        pub struct #name {
+            pub offsets_with_data: ::std::vec::Vec<(u64, ::std::vec::Vec<u8>)>
+        }
+
+        #config_methods
+        #from_impl
+    })
+}
+
+fn generate_configurable_impl(
+    name: &Ident,
```

Comment:
I keep forgetting what `name` refers to. Maybe call it `configurable_struct_name` or something like that?